pkg/apis/core/types.go

```diff
@@ -2053,6 +2080,12 @@ type Container struct {
 	// Compute resource requirements.
 	// +optional
 	Resources ResourceRequirements
+	// Node compute resources allocated to the container.
+	// +optional
+	ResourcesAllocated ResourceList
```

Comment:
I am still confused why this is spec?

Reply:
This is very much alive - I'm a couple of weeks away from having a mostly complete implementation out for review, and also updated API code to move some of the validation earlier into admission stage.

NoRestart/RestartContainer above is a guidance, and lets us decide whether to stop/start container or just call UpdateContainerResources API. The runtime may still choose to restart - we don't have control over it, and no explicit documentation telling runtime not to restart.

ResourcesAllocated was added to Spec to track resources that node has agreed reserve for the pod, and make this info visible to the user. 

We had discussed the alternative of storing this information locally on the node, but I was told that Borg had encountered issues with such approach, and they want to [avoid checkpointing on the node](https://github.com/kubernetes/enhancements/pull/686#discussion_r305090556), and thought it best to have Spec as the source of truth. We had [also looked at relying on PodStatus](https://github.com/kubernetes/enhancements/pull/686#discussion_r281825929), but the Status, per API conventions, isn't allowed store any state that cant be reconstructed through observations.